### PR TITLE
netpacket: ethernet header padding fix

### PR DESCRIPTION
**Pull request by ziemleszcz**
DONE: DTR-94

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imx6ull).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: port/netpacket.c [hunk 1]
```diff
@@ -242,15 +242,27 @@ void netpacket_linkoutput(struct netif *netif, struct pbuf *p)
...
-			/* remove ethernet header from pbuf */
-			pbuf_header(p, -(s16_t)sizeof(struct eth_hdr));
+			/* remove ethernet header */
+			pbuf_remove_header(p, sizeof(struct eth_hdr));
+		}
+		else {
+#if ETH_PAD_SIZE
+			/* remove ethernet header padding */
+			pbuf_remove_header(p, ETH_PAD_SIZE);
+#endif
```
[the remark this answers is not in the document]
Comment (ziemleszcz) [reply port/netpacket.c:252]: Suggestion acknowledged but this code fragment is repeated in other places and I prefer to keep minimal changes in this commit.

File: port/netpacket.c [hunk 2]
```diff
@@ -242,15 +242,27 @@ void netpacket_linkoutput(struct netif *netif, struct pbuf *p)
 		SMEMCPY(&src_addr, netif->hwaddr, sizeof(struct eth_addr));
 
 		if (pcb->type == SOCK_DGRAM) {
-			/* remove ethernet header from pbuf */
-			pbuf_header(p, -(s16_t)sizeof(struct eth_hdr));
+			/* remove ethernet header */
+			pbuf_remove_header(p, sizeof(struct eth_hdr));
```
Comment (agkaminski) [inline port/netpacket.c:246]: Just noticed... Why there was "-(s16_t)sizeof(struct eth_hdr)"? This is substantial change (now it's straight sizeof), is this ok?

Comment (ziemleszcz) [reply port/netpacket.c:246]: It OK, pbuf_header(p, -x) calls pbuf_remove_header(p, x). I just wanted to make it clear we are removing the header here.

Comment (agkaminski) [reply port/netpacket.c:246]: Oh, ok. What a weird API

**Review by agkaminski — APPROVED** [review]

**Pull request merged**